### PR TITLE
Explore tx 3

### PR DESCRIPTION
**Pull request by Jibz1**
Added transaction status on recent transactions 
Gas use is now active
Switched to getTransactionWithEffects
Tx detail page shows the transaction status and failed status message.

Screen Shot 2022-05-02 at 6 06 55 PM
Screen Shot 2022-05-02 at 12 23 22 PM

### CODE REVIEW

File: explorer/client/src/pages/transaction-result/TransactionResult.tsx [hunk 1]
```diff
@@ -39,6 +49,24 @@ const initState: TxnState = {
...
+    const txGasObj = Object.values(txStatusData)[0];
+    const txGas =
+        txGasObj.gas_cost.computation_cost +
+        txGasObj.gas_cost.storage_cost -
+        txGasObj.gas_cost.storage_rebate;
+
+    return {
+        istxSucces,
+        txGas,
+        //  txErr: txStatusData.Failure || '',
```
Comment (666lcz) [inline explorer/client/src/pages/transaction-result/TransactionResult.tsx:68]: nit: remove

Comment (666lcz) [inline explorer/client/src/pages/transaction-result/TransactionResult.tsx:68]: nit: remove

File: explorer/client/src/components/transaction-card/RecentTxCard.tsx [hunk 2]
```diff
@@ -51,10 +71,17 @@ const getRecentTransactions = async (txNum: number) => {
                             res.data
                         )?.recipient;
 
+                        // TODO use ExecutionStatus and add a method on sui.js to get the gas data
```
Comment (666lcz) [inline explorer/client/src/components/transaction-card/RecentTxCard.tsx:74]: can we do this in this PR? Let's do a huddle if you need help

**Review by 666lcz — APPROVED** [review]
Looks great. Requested change to put some common functions in the SDK

File: explorer/client/src/pages/transaction-result/TransactionResult.tsx [hunk 3]
```diff
@@ -39,6 +49,24 @@ const initState: TxnState = {
         epoch: 0,
         signatures: [],
     },
+    txSuccess: false,
+    gasFee: 0,
+    txError: '',
+};
+
+const getGasFeesAndStatus = (txStatusData: ExecutionStatus) => {
```
[the remark this answers is not in the document]
Comment (Jibz1) [reply explorer/client/src/pages/transaction-result/TransactionResult.tsx:57]: I am working on moving it to the helper functions

**Comment by Jibz1** [comment]
Gas payment added, 
Prevent empty search 
added getExecutionStatusType and ExecutionStatusType
some style changes
@666lcz 
Screen Shot 2022-05-02 at 11 39 54 PM

File: explorer/client/src/pages/transaction-result/TransactionResult.tsx (hunk 1 above)
Comment (Jibz1) [reply explorer/client/src/pages/transaction-result/TransactionResult.tsx]: Done

Comment (Jibz1) [reply explorer/client/src/pages/transaction-result/TransactionResult.tsx]: done

**Review by 666lcz — APPROVED** [review]

**Pull request merged**